File: Data structures/Array/Learning/1D Array/GFG_Check_if_two_arrays_are_equal_or_not.py
# https://practice.geeksforgeeks.org/problems/check-if-two-arrays-are-equal-or-not3847/1?

class Solution1:
    # Function to check if two arrays are equal or not.
    def check(self, A, B, N):
        A.sort(), B.sort()

        for i in range(N):
            if (A[i] != B[i]):
                return False

        return True

File: Data structures/Array/Learning/1D Array/GFG_Check_if_two_arrays_are_equal_or_not.cpp
/**
 * @file        GFG_Check_if_two_arrays_are_equal_or_not.cpp
 * @author      Subodh Chandra Shil
 * @link        https://practice.geeksforgeeks.org/problems/check-if-two-arrays-are-equal-or-not3847/1?
 */

#include <bits/stdc++.h>
using namespace std;
#define ll long long

class Solution
{
public:
    bool check(vector<ll> A, vector<ll> B, int N)
    {
        if (A.size() != B.size())
            return 0;

        int n = A.size();
        unordered_map<int, int> elementOccurrences;
        for (int i = 0; i < n; ++i)
        {
            elementOccurrences[A[i]]++;
            elementOccurrences[B[i]]--;
        }

        for (auto i : elementOccurrences)
        {
            if (i.second != 0)
                return false;
        }

        return true;
    }
};

int main()
{

    return 0;
}